src/Lib/Managers/ItemsManager.vala

```diff
@@ -51,33 +49,35 @@ public class Akira.Lib.Managers.ItemsManager : Object {
         window.event_bus.change_item_z_index.connect (on_change_item_z_index);
     }
 
-    public Models.CanvasItem? insert_item (Gdk.EventButton event) {
+    public Models.CanvasItem? insert_item (double x, double y) {
         udpate_default_values ();
 
         Models.CanvasItem? new_item;
         Models.CanvasArtboard? artboard = null;
 
-        foreach (var _artboard in artboards) {
-            if (_artboard.is_inside (event.x, event.y)) {
+        root = window.main_window.main_canvas.canvas.get_root_item ();
```

Comment:
Why did you move this here? Isn't the root canvas always gonna be the same?
Moving it from the construct will set the variable every time a new item is inserted.

Reply:
This is a weird problem. 

Since we moved the `items_manager` initialization at the Window level, the problem now is that we need to have the `canvas` properly set to get the `root` item. 

But the `canvas` itself, in turn, does need an instance of `items manager`, which leads to a recursive import problem: `canvas` need `items_manager` and at the same time `items_manager` needs the `canvas` at init time to get the root item. 

You can see this issue in the `window` construct code, I post it here:
```vala
    construct {
        accel_group = new Gtk.AccelGroup ();
        add_accel_group (accel_group);

        event_bus = new Akira.Services.EventBus ();
        items_manager = new Akira.Lib.Managers.ItemsManager (this);
        action_manager = new Akira.Services.ActionManager (app, this);
        file_manager = new Akira.FileFormat.FileManager (this);
        headerbar = new Akira.Layouts.HeaderBar (this);
        main_window = new Akira.Layouts.MainWindow (this);
        dialogs = new Akira.Utils.Dialogs (this);

        build_ui ();

        move (settings.pos_x, settings.pos_y);

        show_app ();

        main_window.main_canvas.canvas.focus_canvas ();
        event_bus.file_edited.connect (on_file_edited);
        event_bus.file_saved.connect (on_file_saved);
    }
```

## Possible solutions:
1. We could call a method in `items_manager` _after_ both `canvas` and `items_manager` have been created to save the root item of the `canvas` inside the `items_manager`. 
1. We can leave it as is now and simply request the `root` item at every item's creation.
 1. Request `root` item _only_ if it is null (so the first time we create an object). 

What do you think it's the best approach? Is the explanation clean enough? Does it make sense to you?

Reply:
Method 3 seems the easiest and most linear, go for it and thanks for the explanation, that makes sense.

Reply:
Ok perfect!